Source/WebCore/platform/graphics/mac/controls/ControlMac.mm

```diff
@@ -211,8 +249,15 @@ static void drawCellOrFocusRing(GraphicsContext& context, const FloatRect& rect,
             // For slider cells, draw only the knob.
             [(NSSliderCell *)cell drawKnob:rect];
         } else {
-            [cell drawWithFrame:rect inView:view];
-            [cell setControlView:nil];
+            // Check if we can draw the NSCell without the fake NSView.
+            if ([NSCell instancesRespondToSelector:@selector(_setFallbackBackingScaleFactor:)]) {
+                // Don't flip the context if we are just drawing an image.
+                bool flipContext = m_owningPart.type() != StyleAppearance::Checkbox
```

Comment:
Can this logic also be removed once AppKit has fixed rdar://105194487?

Also, it would be nice to put this in a static method.

Reply:
I moved it to a static method called `shouldFlipContext()`.